spec/ameba/rule/lint/useless_assign_spec.cr

```diff
@@ -419,6 +419,35 @@ module Ameba::Rule::Lint
         )
         subject.catch(s).should be_valid
       end
+
+      it "doesn't report if this is a record declaration" do
+        s = Source.new %(
+          record Foo, foo = "foo"
+        )
+        subject.catch(s).should be_valid
+      end
+
+      it "does not report if assignment is referenced after the record declaration" do
+        s = Source.new %(
+          foo = 2
+          record Bar, foo = 3 # foo = 3 is not parsed as assignment
+          puts foo
+        )
+        subject.catch(s).should be_valid
+      end
+
+      it "reports if assignment is not referenced after the record declaration" do
+        s = Source.new %(
+          foo = 2
+          record Bar, foo = 3
+        ), "source.cr"
+        subject.catch(s).should_not be_valid
+        s.issues.size.should eq 1
+
+        issue = s.issues.first
+        issue.location.to_s.should eq "source.cr:2:11"
```

Comment:
The line number is actually correct, but it is hard to follow why this is in 2:11. Why don't you use heredoc instead of a regular string literal? This makes it much easier.

```cr
source = Source.new <<-'SOURCE',
  foo = 2
  record Bar, foo = 3
  SOURCE
  "source.cr"
# ...
issue.location.to_s.should eq "source.cr:1:1"
```

Reply:
This is a convention I follow across the project. Heredoc syntax actually looks weird itself and I don't want to use it here.

Reply:
Well, I think `source.cr:2:11` looks super weird. If you don't want to use heredoc, fine. But maybe you can make this less weird in a different way...

Reply:
Okay. This is a topic of a different PR.